totem/blog/mobile_api.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from typing import List, Optional
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from ninja import Field, FilterSchema, ModelSchema, Router, Schema
+
+from totem.users.models import User
+from .models import BlogPost
+
+from ninja import ModelSchema
+
+class AuthorSchema(ModelSchema):
+    class Meta:
+        model = User
+        fields = ["name", "slug"]
+
+
+class BlogPostSchema(ModelSchema):
+    author: Optional[AuthorSchema] = None
+    header_image_url: Optional[str] = None
+
+    @staticmethod
+    def resolve_header_image_url(obj) -> Optional[str]:
+        if obj.header_image and hasattr(obj.header_image, 'url'):
+            return obj.header_image.url
+        return None
+
+    class Meta:
+        model = BlogPost
+        fields = ["title", "subtitle", "content", "date_published", "slug", "author", "publish"]
+
+blog_router = Router()
+
+@blog_router.get("/posts", response={200: List[BlogPostSchema]}, tags=["blog"], url_name="list_posts")
+def list_posts(request):
```

Comment:
Two things: we'll likely want to paginate this endpoint (see https://django-ninja.dev/guides/response/pagination/).

Also, we may want to make a different schema here to give just the needed info for the list screen, a ListBlogSchema or similar so we're not rendering and sending over all the data at once that the user is unlikely to ever see.

Reply:
Also, with pagination, we'll want the DB to return posts in reverse date order: https://docs.djangoproject.com/en/5.2/ref/models/querysets/#django.db.models.query.QuerySet.order_by